pkg/webhook/webhook_logger.go

```diff
@@ -0,0 +1,69 @@
+// Copyright © 2023 Bank-Vaults Maintainers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"github.com/slok/kubewebhook/v2/pkg/log"
+)
+
+var _ log.Logger = &whLogger{}
+
+type whLogger struct {
+	logger *slog.Logger
+}
+
+// NewWhLogger returns a new log.Logger for a slog implementation.
+func NewWhLogger(l *slog.Logger) log.Logger {
+	return whLogger{l}
+}
+
+func (l whLogger) Infof(format string, args ...interface{}) {
+	l.logger.Info(fmt.Sprintf(format, args...))
+}
+
+func (l whLogger) Warningf(format string, args ...interface{}) {
+	l.logger.Warn(fmt.Sprintf(format, args...))
+}
+
+func (l whLogger) Errorf(format string, args ...interface{}) {
+	l.logger.Error(fmt.Sprintf(format, args...))
+}
+
+func (l whLogger) Debugf(format string, args ...interface{}) {
+	l.logger.Debug(fmt.Sprintf(format, args...))
+}
+
+func (l whLogger) WithValues(kv log.Kv) log.Logger {
+	for k, v := range kv {
+		l.logger.With(slog.Any(k, v))
+	}
+	return l
+}
+
+func (l whLogger) WithCtxValues(ctx context.Context) log.Logger {
+	ctxValues := log.ValuesFromCtx(ctx)
+	for k, v := range ctxValues {
+		l.logger.With(slog.Any(k, v))
+	}
+	return l
+}
+
+func (l whLogger) SetValuesOnCtx(parent context.Context, values log.Kv) context.Context {
+	return log.CtxWithValues(parent, values)
+}
```

Comment:
Two questions here:
1. Why do we need a separate logger interface for a webhook logger?
2. Should these methods clone the original `*slog.Logger` rather than updating the internal one? This can lead to undesirable effects, considering e.g.
```golang
// I want to use the original logger
logger.Infof("message") // prints "message"

// I want a logger with a sticky label
stickyLogger := logger.WithValues(...)
stickyLogger.Infof("message") // prints "message" + sticky label

// I want to use the original logger again, but what is going to happen?
logger.Infof("message") // what will this print?
```

Reply:
1. We use `slok/kubewebhook` for creating the webhooks which needs a `github.com/slok/kubewebhook/v2/pkg/log.Logger`, so whLogger here is wrapper around `*slog.Logger` for this package to use, similar to the `logrus` one that is [implemented within the package](https://github.com/slok/kubewebhook/tree/master/pkg/log/logrus).
2. Thanks for pointing out, I missed this implication!